modmesh/pilot/_linear_wave.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) 2025, Ting-Yu Chuang <tychuang.cs10@nycu.edu.tw>
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# - Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+# - Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+# - Neither the name of the copyright holder nor the names of its contributors
+#   may be used to endorse or promote products derived from this software
+#   without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+
+import numpy as np
+
+from ._base_app import QuantityLine, SolverConfig, OneDimBaseApp
+from ..onedim.linear_scalar import LinearScalarSolver
+
+
+class LinearWave:
+
+    def __init__(self):
+        self.svr = None
+        self.coord_field = None
+        self.wave_field = None
+
+    def build_numerical(self, xmin, xmax, ncelm, cfl):
+
+        # Initialize the solver.
+        self.svr = LinearScalarSolver(xmin, xmax, ncelm, cfl)
+
+        # Setup the solver for time-marching.
+        self.svr.setup_march()
+
+    def build_field(self):
+        self.coord_field = self.svr.xctr() / np.pi
+        self.wave_field = self.svr.get_so0(0).ndarray
```

Comment:
The `LinearWave` class is currently defined in `modmesh.pilot._linear_wave`, following the pattern of `BurgersEquation` in `modmesh.pilot._burgers1d`.

However, I noticed that `ShockTube` is defined directly in `modmesh.onedim.euler1d` and later imported into `modmesh.pilot.euler1d`.

Would it be better to move `LinearWave` into `modmesh.onedim` as well, or is it fine to keep it here in `pilot`?